server/container/util/src/main/java/org/apache/james/util/concurrent/NamedThreadFactory.java

```diff
@@ -19,35 +19,30 @@
 package org.apache.james.util.concurrent;
 
 import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.atomic.AtomicLong;
 
-/**
- * ThreadPool which use name and a counter for thread names
- */
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
 public class NamedThreadFactory implements ThreadFactory {
 
     public final String name;
-    private final AtomicLong count = new AtomicLong();
-    private final int priority;
+    private final ThreadFactory threadFactory;
 
-    public NamedThreadFactory(String name, int priority) {
-        if (priority > Thread.MAX_PRIORITY || priority < Thread.MIN_PRIORITY) {
-            throw new IllegalArgumentException("Priority must be <= " + Thread.MAX_PRIORITY + " and >=" + Thread.MIN_PRIORITY);
-        }
-        this.name = name;
-        this.priority = priority;
+    public static NamedThreadFactory withClassName(Class<?> clazz) {

```

Comment:
having another method of `withObject(Object object)` could save many `getClass()` calls in some where around.

Reply:
but I think that it would not be as clear as waiting for a Class